src/client/Client.cc

```diff
@@ -10974,13 +10988,7 @@ int Client::ll_opendir(Inode *in, int flags, dir_result_t** dirpp,
       return r;
   }
 
-  int r = 0;
-  if (vino.snapid == CEPH_SNAPDIR) {
-    *dirpp = new dir_result_t(in);
-  } else {
-    r = _opendir(in, dirpp, uid, gid);
-  }
-
+  int r = _opendir(in, dirpp, uid, gid);
```

Comment:
This looks fine to me, but can you explain why it's okay to reroute the snapshotted directories through _opendir, with the extra baggage that implies?